File: worker.go
package main

import (
	"fmt"
	"os"
	"sync"
	"time"
)

func worker(id int, tasks <-chan Task, wg *sync.WaitGroup, mu *sync.Mutex, file *os.File) {
	defer wg.Done()
	for task := range tasks {
		fmt.Printf("[Worker %d] Processing %s\n", id, task.ID)
		time.Sleep(500 * time.Millisecond)

		mu.Lock()
		_, err := file.WriteString(task.ID + " processed\n")
		mu.Unlock()

		if err != nil {
			fmt.Printf("[Worker %d] Error: %v\n", id, err)
		} else {
			fmt.Printf("[Worker %d] Completed %s\n", id, task.ID)
		}
	}
}

File: Worker.java
import java.io.FileWriter;
import java.io.IOException;

public class Worker implements Runnable {
    private final SharedQueue queue;
    private final FileWriter writer;

    public Worker(SharedQueue queue, FileWriter writer) {
        this.queue = queue;
        this.writer = writer;
    }

    @Override
    public void run() {
        try {
            while (true) {
                String task = queue.getTask();
                if (task == null) break;

                LoggerUtil.log("Processing task: " + task);
                Thread.sleep(500); // simulate delay

                synchronized (writer) {
                    writer.write(task + " processed\n");
                }
                LoggerUtil.log("Completed task: " + task);
            }
        } catch (InterruptedException | IOException e) {
            LoggerUtil.log("Error: " + e.getMessage());
        }
    }
}
